src/utils/decode-media-data.ts

```diff
@@ -0,0 +1,37 @@
+import { title } from 'case'
+
+import { decodeData as decodeString } from '~/utils/decode-data'
+import type { SupportedMediaType } from '~/constants/media'
+import { IMAGE } from '~/constants/media'
+import type { Media } from '~/models/media'
+
+/**
+ * This interface is a subset of `Media` that types dictionaries sent by the API
+ * being decoded in the `decodeMediaData` function.
+ */
+interface ApiMedia extends Omit<Media, 'frontendMediaType' | 'title'> {
+  title?: string
+}
+
+/**
+ * For any given media, decode the media title, creator name and individual tag
+ * names. Also populates the `frontendMediaType` field on the model.
+ *
+ * @param media - the media object of which to decode attributes
+ * @param mediaType - the type of the media
+ * @returns the given media object with the text fields decoded
+ */
+export const decodeMediaData = (
+  media: ApiMedia,
+  mediaType: SupportedMediaType = IMAGE
+): Media => ({
+  ...media,
+  frontendMediaType: mediaType,
+  title: decodeString(media.title) || title(mediaType),
+  creator: decodeString(media.creator),
+  // TODO: remove `?? []`
+  tags: (media.tags ?? []).map((tag) => ({
```

Comment:
Can the tag name ever be non-string? Would it break in decodeString?

Reply:
I don't think it can, that's handled in the API itself:

https://github.com/WordPress/openverse-api/blob/188b6a0c299a4b79e205c997378e230bf1f96d14/api/catalog/api/models/models.py#L45